src/components/coin.ts

```diff
@@ -269,3 +274,191 @@ export const applyBonusByUserId = async (userId: string): Promise<boolean> => {
+    transferId: string,
+    transferState: TransferState,
+  ) {
+    this.channelId = channelId;
+    this.state = transferState;
+    this.client = client;
+    this.transferId = transferId;
+  }
+
+  // called if state is (believed to be) no longer pending. Transfers coins and updates balances if transfer is confirmed
+  async handleTransaction(): Promise<void> {
+    if (this.state.result === TransferResult.Confirmed) {
+      // Adjust the receiver balance with coins transferred
+      await adjustCoinBalanceByUserId(
+        this.state.receiver.id,
+        this.state.amount,
+        UserCoinEvent.AdminCoinAdjust,
+        <string>(this.state.reason ?? ''),
+        this.client.user?.id,
+      );
+
+      // Adjust the sender balance with coins transferred
+      await adjustCoinBalanceByUserId(
+        this.state.sender.id,
+        <number>(-1 * this.state.amount),
+        UserCoinEvent.AdminCoinAdjust,
+        <string>(this.state.reason ?? ''),
+        this.client.user?.id,
+      );
+    }
+  }
+
+  public getEmbedColor(): ColorResolvable {
+    switch (this.state.result) {
+      case TransferResult.Confirmed:
+        return 'GREEN';
+      case TransferResult.Rejected:
+        return 'RED';
+      default:
+        return 'YELLOW';
+    }
+  }
+
+  public async getStatusAsString(): Promise<string> {
+    switch (this.state.result) {
+      case TransferResult.Confirmed:
+        const newReceiverBalance = await getCoinBalanceByUserId(this.state.receiver.id);
+        const newSenderBalance = await getCoinBalanceByUserId(this.state.sender.id);
+        return `${this.state.receiver.username} accepted the transfer. ${
+          this.state.receiver.username
+        } now has ${newReceiverBalance} Codey ${pluralize(
+          'coin',
+          newReceiverBalance,
+        )} ${getCoinEmoji()}. ${
+          this.state.sender.username
+        } now has ${newSenderBalance} Codey ${pluralize(
+          'coin',
+          newReceiverBalance,
+        )} ${getCoinEmoji()}.`;
+      case TransferResult.Rejected:
+        return `This transfer was rejected by ${this.state.receiver.username}.`;
+      case TransferResult.Pending:
+        return 'Please choose whether you would like to accept this transfer.';
+      default:
+        return `Something went wrong! ${getEmojiByName('codey_sad')}`;
+    }
+  }
+
+  public async getTransferResponse(): Promise<SapphireMessageResponse> {
+    const embed = new MessageEmbed()
+      .setColor(this.getEmbedColor())
+      .setTitle('Coin Transfer')
+      .setDescription(
+        `
+Amount: ${this.state.amount} ${getCoinEmoji()}
+Sender: ${this.state.sender.username}
+
+${await this.getStatusAsString()}
+`,
+      );
+    // Buttons
+    const row = new MessageActionRow().addComponents(
+      new MessageButton()
+        .setCustomId(`transfer-check-${this.transferId}`)
+        .setLabel(getEmojiFromSign(TransferSign.Accept))
+        .setStyle('SECONDARY'),
```

Comment:
I think for this we should set the success style to be SUCCESS And the failure style to be FAILURE. Then we can just change the button contents to be "Accept" and "Reject" respectively.